connectionbuddy/src/main/java/com/zplesac/connectionbuddy/ConnectionBuddy.java

```diff
@@ -339,29 +336,16 @@ private void testNetworkRequest(@NonNull final NetworkRequestCheckListener liste
             @Override
             public void run() {
                 try {
-                    HttpURLConnection httpURLConnection = (HttpURLConnection)
-                        (new URL(NETWORK_CHECK_URL).openConnection());
-                    httpURLConnection.setRequestProperty(HEADER_KEY_USER_AGENT, HEADER_VALUE_USER_AGENT);
-                    httpURLConnection.setRequestProperty(HEADER_KEY_CONNECTION, HEADER_VALUE_CONNECTION);
-                    httpURLConnection.setConnectTimeout(CONNECTION_TIMEOUT);
-                    httpURLConnection.connect();
-
-                    if (httpURLConnection.getResponseCode() == HttpURLConnection.HTTP_NO_CONTENT
-                        && httpURLConnection.getContentLength() == 0) {
-                        callbackExecutor.execute(new Runnable() {
-                            @Override
-                            public void run() {
-                                listener.onResponseObtained();
-                            }
-                        });
-                    } else {
-                        callbackExecutor.execute(new Runnable() {
-                            @Override
-                            public void run() {
-                                listener.onNoResponse();
-                            }
-                        });
-                    }
+                    Socket socket = new Socket();
+                    SocketAddress socketAddress = new InetSocketAddress(NETWORK_CHECK_URL_GOOGLE_DNS, DNS_PORT);
+                    socket.connect(socketAddress, CONNECTION_TIMEOUT);
+                    socket.close();
+                    callbackExecutor.execute(new Runnable() {
```

Comment:
What happens here if the API call is not successfully executed? Does Socket always returns IOException in this case?

Could you maybe provide metrics for this? I'm interested in profiling how much is this implementation faster than the previous one.

Reply:
Here is a comparison from Network Profiler: 

 first the old `HttpUrlConnection` implementation:

<img width="466" alt="Screenshot_2020-10-14 11 36 16_LHnXeA" src="https://user-images.githubusercontent.com/6293174/95971864-0a4dc080-0e12-11eb-8474-0140baae8a9b.png">

and then the `Socket` implementation:

<img width="420" alt="Screenshot_2020-10-14 11 35 53_1N0pRw" src="https://user-images.githubusercontent.com/6293174/95971880-0f127480-0e12-11eb-8e83-e75eb31d0444.png">

The size of the payload is noticeably smaller in the new implementation and the speed is also visibly better.

As for the situation where the API call is not successfully executed, it does return IOException. It's available on all Android API levels as it's a standard Java class.

Also, this implementation is quite popular on StackOverflow :) https://stackoverflow.com/questions/1560788/how-to-check-internet-access-on-android-inetaddress-never-times-out

Reply:
Wow, nice, I really dig this.
Could you also add to the method documentation that the network request is executed on a separate, non-UI thread?

Reply:
Sure, I have updated the method docs.